proving_system/stark/tests/integration_tests.rs

```diff
@@ -214,52 +207,47 @@ fn test_prove_cairo_fibonacci() {
     let raw_trace = CairoTrace::from_file(&dir_trace).expect("Cairo trace binary file not found");
     let memory = CairoMemory::from_file(&dir_memory).expect("Cairo memory binary file not found");
 
-    let execution_trace = build_cairo_execution_trace(&raw_trace, &memory);
-
     let proof_options = ProofOptions {
         blowup_factor: 2,
         fri_number_of_queries: 5,
         coset_offset: 3,
     };
 
-    let cairo_air = cairo::CairoAIR::new(proof_options, &execution_trace);
+    let mut cairo_air = cairo::CairoAIR::new(proof_options, &raw_trace);
+    // PC FINAL AND AP FINAL are not computed correctly since they are extracted after padding to
+    // power of two and therefore are zero
+    cairo_air.pub_inputs.ap_final = FieldElement::zero();
+    cairo_air.pub_inputs.pc_final = FieldElement::zero();
 
-    let result = prove(&execution_trace, &cairo_air);
+    let result = prove(&(raw_trace, memory), &cairo_air);
     assert!(verify(&result, &cairo_air));
 }
 
 #[test_log::test]
-fn test_malicious_trace_does_not_verify() {
-    // A valid execution trace is built from the call_func.cairo binary files.
-    let base_dir = env!("CARGO_MANIFEST_DIR");
-    let dir_trace = base_dir.to_owned() + "/src/cairo_vm/test_data/call_func.trace";
-    let dir_memory = base_dir.to_owned() + "/src/cairo_vm/test_data/call_func.mem";
-
-    let raw_trace = CairoTrace::from_file(&dir_trace).unwrap();
-    let memory = CairoMemory::from_file(&dir_memory).unwrap();
-    let execution_trace = build_cairo_execution_trace(&raw_trace, &memory);
-
-    // Get the columns representation of the execution trace
-    let mut exec_trace_cols = execution_trace.cols();
-    // Get the op1 column
-    let mut op1s = execution_trace.cols()[26].clone();
-    // Write an arbitrary value in the first position of the op1 column.
-    op1s[0] = FE::from(666);
-    // Overwrite the modified op1 column into the execution trace columns representation.
-    exec_trace_cols[26] = op1s;
-
-    // Reconstruct the execution trace with this invalid op1 column.
-    let reconstructed_exec_trace = TraceTable::new_from_cols(&exec_trace_cols);
+fn test_prove_rap_fib() {
+    let trace_length = 16;
+    let trace = fibonacci_rap_trace([FE::from(1), FE::from(1)], trace_length);
+    let mut trace_cols = vec![trace[0].clone(), trace[1].clone()];
+    resize_to_next_power_of_two(&mut trace_cols);
+    let power_of_two_len = trace_cols[0].len();
+    let exemptions = 3 + power_of_two_len - trace_length - 1;
 
-    // We create the new Cairo AIR instance with the malicious trace.
-    let proof_options = ProofOptions {
-        blowup_factor: 2,
-        fri_number_of_queries: 5,
-        coset_offset: 3,
+    let context = AirContext {
+        options: ProofOptions {
+            blowup_factor: 2,
+            fri_number_of_queries: 1,
+            coset_offset: 3,
+        },
+        trace_columns: 3,
+        trace_length: trace_cols[0].len(),
+        transition_degrees: vec![1],
```

Comment:
This should be a vector of two elements, `vec![1, 1]`, since we need the transition degrees for each transition. In this case we have the main and the aux transition, both of degree 1.

Reply:
good catch!!